index.rst

```diff
@@ -1,59 +1,367 @@
+The default role is the user's unix username, which provides write access only to ``u/<user>`` collections.
+These guards against careless fingers, not careless brains - we will not attempt to restrict which roles a user can assume *at the butler client level*.
+
+.. note::
+
+   This proposal intentionally makes no mention of the RFC process that is currently in place for ``/datasets`` for Gen2, or the ``/project`` filesystem.
+
+   In practice, RFCs for most modifications - certainly routine ingests or calibration updates - almost never exceed our `Empowerment of DM team members`_ criteria, and asking for sysadmins who are not domain experts to do the actual work both increases friction and increases the chance something will go wrong (or, rather, the chances that if something does go wrong, it is not fixed immediately).
+
+   This proposal makes no objection to having a separate ``/project`` filesystem for files.
+   But using ``/project`` and symlinks as a way to work around restrictions on write access to appropriate subsets of ``/datasets`` is just that - a workaround - and one that makes it more difficult than it ought to be to find things.
+   If filesystem-level controls (or quotas, etc.) really are necessary even for ``/datasets`` (note that we already allow for filesystem-level controls by symlinking raws from ``/lsstdata`` and ``/external-raw``), I think this is something we can tolerate, but regular directory permission controls within ``/datasets`` would be preferable.
+   If we do use symlinks from ``/project``, we should at least ensure that a::
+
+       /datasets/repo/u/<user> -> /project/<user>/.datarepo
+
+   link (or equivalent) is automatically created with the right permissions for all users, and encourage access only via that symlink (hence the hidden directory as a target).
+
+.. _Empowerment of DM team members: https://developer.lsst.io/team/empowerment.html#empowerment-of-dm-team-members>
+
+
+Personal and test-package repositories
+======================================
+
+This proposal is primarily concerned with long-lived, shared data repositories of the sort that will exist not just at NCSA, but at the IDF, SLAC, CCIN2P3, and other major LSST data facilities.
+
+Small repositories (typically backed by SQLite) are also expected to be common, especially for small-scale CI and local development.
+These repositories should follow the same naming patterns whenever possible, but will generally not need as many levels of indirection to guard against future changes or collections, and many of the collections defined here as ``CHAINED`` or ``TAGGED`` collections can instead be safely defined directly as ``RUN`` collections instead.
```

Comment:
I see @kfindeisen asked a similar question, but I wonder whether instead of an alternate table, we can just say "you do not need to have all of the things in table 1 in your test repo"?

For my own use case, I'm thinking of the gen3 repos that get created by jointcal/fgcm from the exports file in testdata_jointcal. Currently that doesn't conform to this proposal very closely, but it could easily be made to do so, with the only real difference being that it wouldn't contain `raw/` or `u/`, and maybe not even `calib/`: only the nominal `camera` is necessary there, so there's no reason to have `calib` that I can see. Running tests could write to `<instrument>/runs/tests` or somesuch? Some guidance here might be good, so that our various CI repos don't all end up with wildly (or even worse, subtly!) different collections.

Reply:
Avoiding "subtly different collections" is exactly why I wanted the guidance to be more concrete.

Reply:
For the most part, I don't think one can just remove rows from table 1 without modifying the remaining rows.

I do agree that a bit more guidance is warranted, and I'm happy to add that.  But I don't think it'll resolve all questions on what to do in any particular test repo, because I'm getting the impression that pretty much each test repo differs from a standard repo in its own special way.